### PR TITLE
BFD-2590: Terraform CCW RIF Role

### PR DESCRIPTION
**Pull request by keithdadkins**
**JIRA Ticket:**
[BFD-2590](https://jira.cms.gov/browse/BFD-2590)

**User Story or Bug Summary:**

Terraform CCW RIF Role (DO NOT MERGE UNTIL TF STATE IS IMPORTED- Keith)
---

### What Does This PR Do?

- Terraforms the IAM Role the CCW team currently uses to read/write to the etl buckets
- Adds some sensitive params for use by the assume role policy
- Removes the deprecated IAM user, groups, access keys, the team used previously
- Refactors the terraform a bit to account for the verification bucket now being permanent
- Updates terraform-docs

I still need to import the existing role into our tf state but waiting on a few deployment issues before doing so, so it's not ready for merging, but it is ready for review otherwise.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* [x] Existing IAM role imported into appropriate tf state

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/terraform/services/pipeline/iam.tf [hunk 1]
```diff
@@ -1,35 +1,42 @@
...
 }
 
-resource "aws_iam_group" "this" {
-  count = local.create_etl_user ? 1 : 0
-  name  = "bfd-${local.env}-${local.legacy_service}"
-  path  = "/"
+resource "aws_iam_role" "ccw_rif" {
+  count = local.is_prod ? 1 : 0
+  name = "bfd-${local.env}-ccw-rif"
+  description = "Role assumed by CCW to read and write to the ${local.env} ETL bucket."
```
Comment (mjburling) [inline ops/terraform/services/pipeline/iam.tf:11]: I think we can continue avoiding comments here if we expand the description just a bit, maybe something like:
```suggestion
  description = "Role assumed by CCW to read and write to the ${local.env} production and verification ETL buckets."
```

File: ops/terraform/services/pipeline/iam.tf
Comment (mjburling) [inline ops/terraform/services/pipeline/iam.tf:1]: Could you remove the outdated TODO comment from the `aws_iam_policy.etl-rw-s3` resource that reads:
> TODO: Remove after CCW Verification is complete, ca Q1 2023.

File: ops/terraform/services/pipeline/s3.tf [hunk 2]
```diff
@@ -72,3 +73,47 @@ resource "aws_s3_bucket_notification" "etl_bucket_notifications" {
...
+
+# CCW Verification Bucket
+resource "aws_s3_bucket" "ccw-verification" {
+  count  = local.is_prod ? 1 : 0
+  bucket = "bfd-prod-ccw-verification"
+  # similar tags to the sole production etl user
+  tags = {
+    Layer   = local.layer,
+    role    = local.legacy_service
+    Note    = "Temporary resource to be removed ca Q1 2023"
```
Comment (mjburling) [inline ops/terraform/services/pipeline/s3.tf:85]: Maybe the `tag:Note` gets updated here? How do you feel about `tag:Note`, anyhow?
```suggestion
    Note    = "Pre-production sensitive data used for CCW and BFD process verification"
```

**Review by mjburling — CHANGES_REQUESTED** [review]
Thanks for putting this together, @keithdadkins!

This is all consistent with the existing solution, and I've only left a few notes on non-functional in-line documentation/comments that I hope you can address soonest–this ought to merge today!

**Review by mjburling — APPROVED** [review]

**Review by brandoncruz3 — APPROVED** [review]

**Pull request merged**